src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -337,109 +332,183 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
                             INTERNAL FUNCTIONS
     *******************************************************************************/
 
-    /// @return numOperatorsPerQuorum is the list of number of operators per quorum in quorumNumberss
+    /** 
+     * @notice Register the operator for one or more quorums. This method updates the
+     * operator's quorum bitmap, socket, and status, then registers them with each registry.
+     */
     function _registerOperator(
-        address operator, 
+        address operatorToRegister, 
+        bytes32 idToRegister,
         bytes calldata quorumNumbers,
-        string memory socket
-    ) internal virtual returns(uint32[] memory, bytes32) {        
-        // Create and validate bitmap from quorumNumbers
-        uint256 quorumBitmap = BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers);
-        require(quorumBitmap <= MAX_QUORUM_BITMAP, "BLSRegistryCoordinatorWithIndices._registerOperator: bitmap exceeds max bitmap size");
-        require(quorumBitmap != 0, "BLSRegistryCoordinatorWithIndices._registerOperator: bitmap cannot be 0");
-        
+        string memory socket,
+        bool performChurn,
```

Comment:
seems like an unnecessary / duplicate parameter?

Reply:
ahh, I guess it enables the clearer 'churn required for overfilled quorum' error below, rather than simply hitting an index out-of-bounds error. seems OK then

Reply:
Yes, also this most closely mimics the current implementation, which performs a loop over each quorum after registration and checks if the new operator count is over the max.

Reply:
I split the churn function back out of this method and into the top-level method. Needed to in order to fix the stack too deep issue. Anyway, I'm not too mad. I think it's a bit easier to understand now.